VideoApp/VideoApp/Video/VideoAppCameraSource.swift

```diff
@@ -0,0 +1,159 @@
+            if #available(iOS 13, *) {
+                builder.orientationTracker = UserInterfaceTracker(scene: UIApplication.shared.keyWindow!.windowScene!)
+            }
+            
+            switch self.appSettingsStore.topology {
+            case .group: builder.rotationTags = .remove
+            case .peerToPeer: break
+            }
+        }
+        
+        cameraSource = CameraSource(options: options, delegate: self)
+        
+        if let cameraSource = cameraSource {
+            localVideoTrack = LocalVideoTrack(source: cameraSource, enabled: true, name: "camera")
+            startCameraSource()
+        }
+    }
+    
+    private func startCameraSource() {
+        guard let cameraSource = cameraSource else { return }
+
+        let targetSize: CMVideoDimensions
+        let cropRatio: CGFloat
+        let frameRate: UInt
+        
+        switch appSettingsStore.videoCodec {
+        case .h264, .vp8:
+            targetSize = CMVideoDimensions(width: 544, height: 480)
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 20
+        case .vp8Simulcast:
+            targetSize = CMVideoDimensions(width: 900, height: 720)
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 24
+        }
+        
+        let captureDevice = CameraSource.captureDevice(position: .front)!
+        let preferredFormat = selectVideoFormatBySize(captureDevice: captureDevice, targetSize: targetSize)
+        preferredFormat.frameRate = min(preferredFormat.frameRate, frameRate)
+        
+        let cropDimensions: CMVideoDimensions
+        
+        if preferredFormat.dimensions.width > preferredFormat.dimensions.height {
+            cropDimensions = CMVideoDimensions(
+                width: Int32(CGFloat(preferredFormat.dimensions.height) * cropRatio),
+                height: preferredFormat.dimensions.height
+            )
+        } else {
+            cropDimensions = CMVideoDimensions(
+                width: preferredFormat.dimensions.width,
+                height: Int32(CGFloat(preferredFormat.dimensions.width) * cropRatio)
+            )
+        }
+        
+        let outputFormat = VideoFormat()
+        outputFormat.dimensions = cropDimensions
+        outputFormat.pixelFormat = preferredFormat.pixelFormat
+        outputFormat.frameRate = 0
+        
+        cameraSource.requestOutputFormat(outputFormat)
+        
+        cameraSource.startCapture(device: captureDevice, format: preferredFormat) { [weak self] _, _, error in
+            guard error == nil else { return }
+            
+            self?.localMediaController.videoCaptureStarted()
+        }
+    }
+        
+    private func selectVideoFormatBySize(captureDevice: AVCaptureDevice, targetSize: CMVideoDimensions) -> VideoFormat {
+        let supportedFormats = Array(CameraSource.supportedFormats(captureDevice: captureDevice)) as! [VideoFormat]
```

Comment:
@paynerc notice the casting I have to do here. It seems NSOrderedSet does not bridge to Swift well. I end up getting a set of type `Any` so I have to cast it to do anything. Maybe the SDK interface should just return `[VideoFormat]`?

Reply:
I *think* an ordered set was returned to remove duplicate formats, but I would need to go back and look at that. That would be a breaking change to the API, so we would need a major version bump if we were to change what it returns.

Reply:
Yeah was just food for thought. It seems like SDK could internally use NSOrderedSet but have the public interface return array and that would work for everyone.

Reply:
Yeah, I was thinking the same thing, but we are unable to make that change until we bump major version.